modin/pandas/series.py

```diff
@@ -1734,6 +1734,20 @@ def _getitem(self, key):
             return self._reduce_dimension(result)
         return self.__constructor__(query_compiler=result)
 
+    @classmethod
+    def _inflate_light(cls, query_compiler, name):
+        return cls(query_compiler=query_compiler, name=name)
+
+    @classmethod
+    def _inflate_full(cls, pandas_series):
+        return cls(data=pandas_series)
+
+    def __reduce__(self):
+        self._query_compiler._modin_frame._materialize()
```

Comment:
I think this leaks too much into the API layer.

Reply:
I've renamed the method to `.finalize()` and added the same method to query compiler level, so API level does not directly access the frame level anymore.